### PR TITLE
Hooked up Views to Riot API Logic 

### PR DESCRIPTION
**Pull request by SpencerReyka**
Hooked up views endpoint (/stats/calculate/) to the Dps Threat Calculate logic

Able to provide the endpoint names in the body, and it will parse and calcuate.

Screenshot 2024-02-06 at 2 43 25 AM

### CODE REVIEW

File: docs/ToDo.md [hunk 1]
```diff
@@ -2,9 +2,12 @@
 
 
 ## High Priority TO-DO
-- add Django 
+- Figure out stats/losses/core code, currently a class in the views
+- Figure out how to create and close a connection for the dps threat lookup
 - DTO stuff, figure out 
 - Create Project folder, and app (project is riot-api, app is stats?)
+- Probably remove the Runner code
```
Comment (claybrooks) [inline docs/ToDo.md:13]: Is the runner code provided by django?  I would have figured django gave you some kind of loop out of the box

File: docs/ToDo.md [hunk 2]
```diff
@@ -2,9 +2,12 @@
...
+- Add to readme about endpoint? Maybe look into swagger too.
```
Comment (claybrooks) [inline docs/ToDo.md]: definitely look in to swagger

File: stats/losses/core.py [hunk 3]
```diff
@@ -10,6 +10,35 @@ def __init__(self, api_key):
         self.api_key = api_key
         print("created Config")  
 
+class StatsService():
+
+    def __init__(self):
+        API_KEY = os.getenv('RIOT_API_KEY_APP')
```
Comment (claybrooks) [inline stats/losses/core.py:16]: you should consider having a layer to hide config sources (enironment, files, command line, key vaults, etc...).  Will make testing easier and give you more flexibility in the long run.

Build the configuration object somewhere else, like with https://pypi.org/project/ConfigArgParse/ or https://github.com/dynaconf/dynaconf, and pass it in to the object here

File: stats/losses/core.py [hunk 4]
```diff
@@ -10,6 +10,35 @@ def __init__(self, api_key):
...
+        self.riotProxy = RiotProxy(API_KEY)
+
+    def calculate_dps_threats(self, names):
```
Comment (claybrooks) [inline stats/losses/core.py:19]: res = [self.calculate_dps_threat(x) for x in names]

File: stats/losses/core.py [hunk 5]
```diff
@@ -10,6 +10,35 @@ def __init__(self, api_key):
...
+
+        for name in names:
+            dps_threat = self.calculate_dps_threat(name)
+            res.append(dps_threat)
+
+        return res
+
+    def calculate_dps_threat(self, name):
+
+        postgresProxy = PostgresProxy()
```
Comment (claybrooks) [inline stats/losses/core.py]: you should make PostgresProxy with'able

```
with PostgresProxy() as f:
    # write the code
```

That way you don't have to worry about calling close, mostly in the case that any code between open and close throws an exception

File: stats/models/match.py [hunk 6]
```diff
@@ -1,4 +1,6 @@
-class Match():
+from django.db import models
+
+class Match(models.Model):
     def __init__(self, username, match_data):
         self.id = match_data["metadata"]["matchId"]
```
Comment (claybrooks) [inline stats/models/match.py:5]: possible key not found exception.  You should try/catch and return in the case the id doesn't exist

File: stats/views.py [hunk 7]
```diff
@@ -1,5 +1,29 @@
...
\ No newline at end of file
+    print(request.body)
+
+    #set_up_code()
+
+    return HttpResponse("Hello, world. You're at the polls index.")
+
+
+def calculate(request):
+    statsService
```
Comment (claybrooks) [inline stats/views.py]: what is this for

File: stats/views.py [hunk 8]
```diff
@@ -1,5 +1,29 @@
...
+    body = json.loads(request.body.decode('utf-8'))
```
Comment (claybrooks) [inline stats/views.py]: this can throw exceptions with bad data

**Review by claybrooks — CHANGES_REQUESTED** [review]
Couple things overall:

1. You should really use pythons typing helpers.  Will make your life easier
2. You should get a logger class in now before it becomes too much to wire in later
3. You should choose a naming scheme for your privates

File: stats/views.py (hunk 7 above)
Comment (SpencerReyka) [reply stats/views.py]: Removed, didn't mean for that to be there.

File: stats/views.py (hunk 8 above)
Comment (SpencerReyka) [reply stats/views.py]: Fixed!

Comment (SpencerReyka) [reply stats/views.py]: Will need to be more conscious of this going forward

File: stats/losses/core.py (hunk 5 above)
Comment (SpencerReyka) [reply stats/losses/core.py]: Oh i forgot all about that! ty ty

File: stats/losses/core.py (hunk 4 above)
Comment (SpencerReyka) [reply stats/losses/core.py:19]: Beautiful!

File: stats/losses/core.py (hunk 3 above)
Comment (SpencerReyka) [reply stats/losses/core.py:16]: Going to add this to the ToDo, will look into it today

File: docs/ToDo.md (hunk 2 above)
Comment (SpencerReyka) [reply docs/ToDo.md]: Added to todo

File: docs/ToDo.md (hunk 1 above)
Comment (SpencerReyka) [reply docs/ToDo.md:13]: This first started out as just a single file script called Runner.py, then i moved the business logic code into another file and left Runner.py as the starting point. This past week I added Django, and then hooked everything up to flow from the Django manage.py script. There is still some old code floating around that needs to be nuked (aka Runner.py)

**Review by claybrooks — APPROVED** [review]
sublime

**Pull request merged**